File: TopInterview150/SlidingWindow/MinimumSizeSubarraySum/solution.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        """
        :type target: int
        :type nums: List[int]
        :rtype: int
        """
        p1 = 0
        minSize = 1000000000
        currSum = 0
        for p2 in range(len(nums)):
            currSum += nums[p2]
            while (currSum >= target):
                minSize = min(minSize, p2-p1+1)
                currSum -= nums[p1]
                p1+=1
        if minSize == 1000000000:
            return 0
        return minSize

File: TopInterview150/SlidingWindow/MinimumSizeSubarraySum/solution.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int p1 = 0;
        int minSize = 1000000000;
        int currSum = 0;
        for (int p2 = 0; p2 < nums.size(); p2++)
        {
            currSum += nums[p2];
            while (currSum >= target)
            {
                minSize = min(minSize, p2-p1+1);
                currSum -= nums[p1];
                p1++;
            }
        }
        if (minSize == 1000000000) return 0;
        return minSize;
    }
};